drivers/pds/targetcluster/targetcluster.go

```diff
@@ -48,6 +48,60 @@ type TargetCluster struct {
 	kubeconfig string
 }
 
+func (targetCluster *TargetCluster) IsLatestPDSHelm(helmChartversion string) (bool, error) {
+	cmd := "helm ls --all -n pds-system | tail -n+2 | awk '{print $9}' "
+	output, _, err := osutils.ExecShell(cmd)
+	if err != nil {
+		return false, err
+	}
+	log.InfoD("Installed PDS Helm chart version - %v", output)
+	_, pdsHelmVersion, _ := strings.Cut(output, "-")
+	_, helmVersion, _ := strings.Cut(pdsHelmVersion, "-")
+	helmVersion = strings.TrimSpace(helmVersion)
+	helmChartversion = strings.TrimSpace(helmChartversion)
+	log.Debugf("Installed PDS Helm version %s and helm chart version passed %s", helmVersion, helmChartversion)
+	return strings.EqualFold(helmVersion, helmChartversion), nil
+}
+
+func (targetCluster *TargetCluster) DeRegisterFromControlPlane() error {
+	pods, err := k8sCore.GetPods(PDSNamespace, nil)
+	if err != nil {
+		return err
+	}
+	if len(pods.Items) > 0 {
+		log.InfoD("Uninstalling PDS from the Target cluster")
+		cmd := fmt.Sprintf("helm uninstall  pds --namespace %v", PDSNamespace)
+		output, _, err := osutils.ExecShell(cmd)
+		if err != nil {
+			return fmt.Errorf("error occured while removing the pds helm chart: %v", err)
+		}
+		log.InfoD("helm uninstall output: %v", output)
+	} else {
+		log.InfoD("No pods are avaialble in the %s namespace", PDSNamespace)
+		cmd := "helm list -A"
+		output, _, err := osutils.ExecShell(cmd)
+		if err != nil {
+			return fmt.Errorf("error occured while listing helm installations: %v", err)
+		}
+		log.InfoD("helm list output: %v", output)
+	}
+	log.Infof("wait till all the pds-system pods are deleted")
+	err = wait.Poll(10*time.Second, 5*time.Minute, func() (bool, error) {
+		pods, err := k8sCore.GetPods(PDSNamespace, nil)
+		if err != nil {
+			return false, nil
+		}
+		if len(pods.Items) == 0 {
+			log.InfoD("pds is uninstalled")
+			return true, nil
+		}
+		log.Infof("There are %d pods present in the namespace %s", len(pods.Items), PDSNamespace)
+		return false, nil
+	})
+
```

Comment:
Remove PV/PVCs as well.

Reply:
We don't need to explicitly remove PVC/PV, it gets synced when new pds helm installation happens.

Reply:
Please add a todo for the same, otherwise, the cluster will be flooded with a lot of PV/PVCs etc.

Reply:
No It will not, 
1. I have verified it, Its not creating a new PVC/PV all the time when we are deregistering and registering the cluster
2. Also, we are using E2E pipeline for this test case and which means we are not reusing the cluster for other testing activities also.